frontend/packages/metal3-plugin/src/features.ts

```diff
@@ -0,0 +1,17 @@
+import * as _ from 'lodash';
+import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
+import { InfrastructureModel } from '@console/internal/models';
+import { getInfrastructurePlatform } from '@console/shared/src/selectors/infrastructure';
+import { BAREMETAL_FLAG } from './constants';
+import { setFlag, handleError } from '@console/internal/actions/features';
+
+export const detectBaremetalPlatform = (dispatch) =>
```

Comment:
Please improve the typings in `packages/console-plugin-sdk/src/typings/features.ts`

```diff
 export interface ActionFeatureFlag {
   /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
-  detect: (dispatch: Dispatch) => Promise<any>;
+  detect: ActionFeatureFlagDetector;
 }

 // ..

+export type ActionFeatureFlagDetector = (dispatch: Dispatch) => Promise<any>;
```

and use it here

```ts
export const detectBaremetalPlatform: ActionFeatureFlagDetector = (dispatch) =>
```